test/e2e/utils/pod/pod.go

```diff
@@ -299,6 +299,28 @@ func nfdWorkerSpec(opts ...SpecOption) *corev1.PodSpec {
 				},
 			},
 		},
+		Tolerations: []corev1.Toleration{
+			{
+				Key:    "nfd.node.kubernetes.io/fake-special-node",
+				Value:  "exists",
+				Effect: "NoExecute",
+			},
+			{
+				Key:    "nfd.node.kubernetes.io/fake-dedicated-node",
+				Value:  "true",
+				Effect: "NoExecute",
+			},
+			{
+				Key:    "nfd.node.kubernetes.io/performance-optimized-node",
+				Value:  "true",
+				Effect: "NoExecute",
+			},
+			{
+				Key:    "nfd.node.kubernetes.io/foo",
+				Value:  "true",
+				Effect: "NoExecute",
+			},
+		},
```

Comment:
Hmm, shouldn't we add these to nfd-master, too?

Also, I'd suggest to not hardcode these but add a `SpecWithTolerations()` "option function" similar to what we do with other podspec customization. WDYT?

Reply:
Good point, thanks. Done.